File: Practise/DOTW.cs
/* COPYRIGHT (C) HARRY CLARK */

/* CS1_OOP GENERAL JAVA HELP */

/* AS A SLAP IN THE FACE WITH REGARDS TO JAVA AS A LANGUAGE, ALLOW ME TO RE-WRITE MY JAVA DOTW */
/* PROGRAM IN C# */

/* BECAUSE JAVA'S SUPPOSED "SURROGATE" MEANS OF METHOD MODIFIERS AND ACCESSORS */
/* GIVE ME AN ANEURYSM AND SERVE TO PROVE HOW MUCH OF A HEADACHE OOD IS IN GENERAL */

/* THE FOLLOWING FILE SERVES TO SHOWCASE HOW TO WORK WITH USER INPUTS */

using System;

public static class DOTW
{
    public static int USER_INPUT;

    public enum DAYS
    {
        SUNDAY,
        MONDAY,
        TUESDAY,
        WEDNESDAY,
        THURSDAY,
        FRIDAY,
        SATURDAY,
    }

    public static DAYS DAY_OF_WEEK(int USER_INPUT)
    {
        DAYS DAY;

        switch (USER_INPUT)
        {
            case 1:
                DAY = DAYS.SUNDAY;
                break;

            case 2:
                DAY = DAYS.MONDAY;
                break;

            case 3:
                DAY = DAYS.TUESDAY;
                break;

            case 4:
                DAY = DAYS.WEDNESDAY;
                break;

            case 5:
                DAY = DAYS.THURSDAY;
                break;

            case 6:
                DAY = DAYS.FRIDAY;
                break;

            case 7:
                DAY = DAYS.SATURDAY;
                break;

            default:
                throw new ArgumentException("Invalid Input. Please enter a number between 1 and 7.");
        }

        return DAY;
    }
}

public static class Program
{
    private static void Main()
    {
        Console.Write("Enter a number between 1 and 7: ");
        DOTW.USER_INPUT = int.Parse(Console.ReadLine());
        DOTW.DAYS DOW;

        /* USE A TRY CATCH TO BE ABLE TO DETERMINE THE UNSATISFACTORY */
        /* PASSED THROUGH THE PROGRAM */

        try
        {
            DOW = DOTW.DAY_OF_WEEK(DOTW.USER_INPUT);
            Console.WriteLine($"The day of the week is: {DOW}");
        }
        catch (ArgumentException EXEC)
        {
            Console.WriteLine($"Error: {EXEC.Message}");
        }
    }
}

File: Practise/DOTW.java
/* COPYRIGHT (C) HARRY CLARK */

/* CS1_OOP GENERAL JAVA HELP */

/* BECAUSE JAVA GIVES ME A HEADACHE, I CONJURED UP A */
/* PROCESS TO BE ABLE TO ENGAGE WITH THE LANGUAGE A BIT MORE */

/* THE FOLLOWING FILE SERVES TO SHOWCASE HOW TO WORK WITH USER INPUTS */

import java.util.Scanner;

public class DOTW 
{
    public static final class METHODS
    {
        private static final int DAY = 0;
        public static final Scanner USER_IO = new Scanner(System.in);
    }


    public static void SWITCH_DAY()
    {
        switch (METHODS.DAY) {
            case 1:
                System.out.println("Today is Monday");
                break;

            case 2:
                System.out.println("Today is Tuesday");
                break;

            case 3:
                System.out.println("Today is Wednesday");
                break;

            case 4:
                System.out.println("Today is Thursday");
                break;

            case 5:
                System.out.println("Today is Friday");
                break;

            case 6:
                System.out.println("Today is Saturday");
                break;

            case 7:
                System.out.println("Today is Sunday");
                break;
        
            default:
                break;
        }
    }
}

class MAIN
{
    public static void main(String[] args)
    {
        DOTW DAY = new DOTW();

        System.out.println("Enter a number between 1 and 7");

        DAY.METHODS.USER_IO();
        DOTW.METHODS.SWITCH_DAY();
    }
}
